nxc/modules/snipped.py

```diff
@@ -0,0 +1,126 @@
+        self.connection = connection
+        self.share = "C$"
+
+        self.hostname = connection.hostname
+        self.domain = connection.domain
+        self.username = connection.username
+        self.password = getattr(connection, "password", "")
+        self.host = connection.host
+        self.kerberos = connection.kerberos
+        self.lmhash = getattr(connection, "lmhash", "")
+        self.nthash = getattr(connection, "nthash", "")
+        self.aesKey = connection.aesKey
+        self.use_kcache = getattr(connection, "use_kcache", False)
+
+        target = Target.create(
+            domain=self.domain,
+            username=self.username,
+            password=self.password,
+            target=self.hostname + "." + self.domain if self.kerberos else self.host,
+            lmhash=self.lmhash,
+            nthash=self.nthash,
+            do_kerberos=self.kerberos,
+            aesKey=self.aesKey,
+            no_pass=True,
+            use_kcache=self.use_kcache,
+        )
+
+        dploot_conn = self.upgrade_connection(target=target, connection=connection.conn)
+
+        output_base_dir = join(NXC_PATH, "modules", "snipped", "screenshots")
+        os.makedirs(output_base_dir, exist_ok=True)
+
+        context.log.debug("Getting all user folders")
+        try:
+            user_folders = dploot_conn.listPath(self.share, "\\Users\\*")
+        except Exception as e:
+            context.log.fail(f"Failed to list user folders: {e}")
+            return
+
+        context.log.debug(f"User folders: {user_folders}")
+        if not user_folders:
+            context.log.fail("No User folders found!")
+            return
+        else:
+            context.log.display("Attempting to download screenshots if existent.")
+
+        for user_folder in user_folders:
+            if not user_folder.is_directory():
+                continue
+            folder_name = user_folder.get_longname()
+            if folder_name in [".", "..", "All Users", "Default", "Default User", "Public"]:
+                continue
+            if self.users and folder_name not in self.users:
+                continue
+
+            screenshot_path = ntpath.normpath(join(r"Users", folder_name, self.screenshot_path_stub))
+            try:
+                screenshot_files = dploot_conn.listPath(self.share, screenshot_path + "\\*")
+            except Exception as e:
+                context.log.debug(f"Screenshot folder {screenshot_path} not found for user {folder_name}: {e}")
+                continue
+
+            if not screenshot_files:
+                context.log.debug(f"No screenshots found in {screenshot_path} for user {folder_name}")
+                continue
+
+            user_output_dir = join(output_base_dir, folder_name)
+            os.makedirs(user_output_dir, exist_ok=True)
+
+            context.log.display(f"Downloading screenshots for user {folder_name}")
+            downloaded_count = 0
+            for file in screenshot_files:
```

Comment:
While testing also the .ini file got downloaded, which should probably be excluded. Also somehow the files have all the size 0 bytes. NetExec has a build in `connection.get_file()` method, which should be used to retrieve these files. Take a look at this as an example: https://github.com/Pennyw0rth/NetExec/blob/0da04ddb20c9c8992a33169489a78786cdc2ffd3/nxc/modules/teams_localdb.py#L21-L25
![image](https://github.com/user-attachments/assets/9797b61e-0251-4e58-b78f-f79c8163e689)

Reply:
Maybe a sanity check for the file size would be good, including an error when the size is 0.

Reply:
Exclustions are now implemented:
https://github.com/Yeeb1/NetExec/blob/af512f1e6ba7dd37f11e430bdf18b0d22347ddd7/nxc/modules/snipped.py#L18

Check for file size is now implemented as well:

https://github.com/Yeeb1/NetExec/blob/af512f1e6ba7dd37f11e430bdf18b0d22347ddd7/nxc/modules/snipped.py#L102

Reply:
Great 👍